pages/api/cancel.ts

```diff
@@ -59,6 +63,42 @@ export default async function handler(req, res) {
     return res.status(403).json({ message: "Cannot cancel past events" });
   }
 
+  const organizer = await prisma.user.findFirst({
+    where: {
+      id: bookingToDelete.userId as number,
+    },
+    select: {
+      name: true,
+      email: true,
+      timeZone: true,
+    },
+  });
+
+  const evt: CalendarEvent = {
+    type: bookingToDelete?.title,
+    title: bookingToDelete?.title,
+    description: bookingToDelete?.description || "",
+    startTime: bookingToDelete?.startTime.toString(),
+    endTime: bookingToDelete?.endTime.toString(),
+    organizer: organizer,
+    attendees: bookingToDelete?.attendees.map((attendee) => {
+      const retObj = { name: attendee.name, email: attendee.email, timeZone: attendee.timeZone };
+      return retObj;
+    }),
+  };
+
+  // Hook up the webhook logic here
+  const eventTrigger = "BOOKING_CANCELLED";
+  // Send Webhook call if hooked to BOOKING.CANCELLED
+  const subscriberUrls = await getSubscriberUrls(
+    bookingToDelete.userId,
+    bookingToDelete.eventTypeId,
+    eventTrigger
+  );
+
+  await Promise.all(
+    subscriberUrls.map((url) => sendPayload(eventTrigger, new Date().toISOString(), url, evt))
+  );
```

Comment:
Same comment as above

Reply:
Just like in case of booking created/rescheduled, the webhook is only checked (and fired) after the cancellation is complete, so it doesn't interfere with that flow. :)